### ISSUE TITLE
In Larva tests, there should be a mechanism to wait for a condition to become true

### ISSUE THREAD

**Issue by mhdirkse** [ibissource/iaf#4739]
**Describe the issue**
This is inspired by Heinenoord. In that application, a long wait during the Larva test was needed for the following reason. The Larva test triggered an adapter that processed files. The adapter stored a message in the message store. Another adapter read the message and called a stub service that was controlled by Larva. The Larva service read the message but did not provide a reply. This way the system-under-test experienced a timeout. It put the message in the error store. The Larva test had to wait until the message was in the error store. Then the Larva test could check that the IBISSTORE table had a record with type "E".

The issue is that we could only implement this test using a fixed waiting time. In modern test tools, there is syntax to wait for the condition to become true, without waiting too long or too short.

I have the following ideas to make a variable waiting time possible:
* We can extend the ManageDatabase and ManageFileSystem adapters to wait until a query produces a desired result or until some condition about the file system becomes true.
* We can extend Larva's syntax to support loops. The loop syntax should incorporate a waiting time and a test for ending the loop. The advantage of this approach is that more variation is possible and that the repetition can be expressed more clearly in the Larva output.

**Reporter**
Martijn Dirkse

**To Reproduce**

**Screenshots**

**Configuration**
```
Please provide the configuration of the Pipe or Receiver with the problem.
```

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
```
FF! 7.8-RC3: Frank2Heinenoord
Running on LAPTOP-R1SK28C using Apache Tomcat/9.0.68
Java Version: OpenJDK Runtime Environment (1.8.0_292-b10)
Heap size: 194M, total JVM memory: 824M
Free disk space: 198GB, total disk space: 464GB
Up since: 2023-04-03 20:08:19 (36m)```

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

**Comment by themidix** [comment]
Hi @mhdirkse,  I'm a developer looking to make a first contribution to Frank!Framework, and I'd like to work on this issue if it's still available.
The async-wait problem you're describing is a common pain point in integration testing, and I think it's worth solving. My plan: add optional waitfor.timeout / waitfor.interval properties to existing Larva actions, so a step polls until the expected result appears or the timeout elapses, additive and backward compatible, rather than new loop syntax in the scenario format.
Before I start, I'd like to be assigned, and I'd appreciate a quick sanity check on that approach vs. the loop-syntax alternative, since it affects public Larva syntax.

**Comment by nielsm5** [comment]
A propery which you can give an expression would be preffered.
E.g. 
```
waitfor.xPath=/xpath/[]
```

See: https://github.com/frankframework/frankframework/issues/10798 
And the proposed next steps https://github.com/frankframework/frankframework/issues/3470#issuecomment-4498820509

**Comment by themidix** [comment]
Good point, @nielsm5 — and thanks for linking #10798. I agree that a fixed full-content match is often too strict for a wait condition (e.g., in the Heinenoord case, you really want "does a row with type=E exist," not a byte-for-byte match against a static expected file). I'll design waitfor around an expression rather than a full compare: something like waitfor.xPath=... / waitfor.jPath=..., evaluated against the actual result on each poll, alongside waitfor.timeout / waitfor.interval for the polling cadence. If no expression is given, it'll fall back to the existing full-result comparison, so nothing breaks for simple cases. This also means waitfor can share the same expression-evaluation path as whatever gets built for #10798, instead of introducing a second one. I'll fold this into the implementation plan before I start coding.

**Comment by themidix** [comment]
Following up on my earlier comments — I've pushed an initial implementation to my fork ([`feature/4739-larva-waitfor-condition`](https://github.com/themidix/frankframework/tree/feature/4739-larva-waitfor-condition)) so there's something concrete to react to, but there's one open question from my first comment that I want to make sure gets answered before I take this further: **should `waitfor` live as step properties on the existing action model, or should it be new loop syntax in the scenario format instead?** @nielsm5's reply helped settle *what* to check (an expression like `waitfor.xPath`, evaluated per poll, rather than only a full-content compare) — but it didn't weigh in on *where* the retry mechanism should live, and that's the part that actually changes public Larva syntax.

What's implemented so far, in case it helps ground the discussion:
- `waitfor.timeout` / `waitfor.interval` / `waitfor.xPath` as step properties, in the same per-step property namespace already used for things like `param1.name`.
- `waitfor.timeout` is rejected at scenario-load time unless the step's target action is backed by `FixedQuerySender` or `DelaySender` — those are the only action types whose read can be safely re-executed on every poll attempt; others either throw on a second read or would destructively consume a new message from a queue.
- `waitfor.xPath` reuses the existing `TransformerPool`/`IfPipe` expression machinery, so it should stay consistent with whatever #10798 ends up using.
- When `waitfor.xPath` is set, it only decides *when to stop polling* — the step's final pass/fail is still the normal full-content diff against the expected file, so no scenario's existing verification gets weaker.
- Scope of the first PR is deliberately narrow: `waitfor.xPath` only (no `waitfor.jPath` yet), and only for query-style reads. I'd rather get the mechanism right in a small PR than land a big one that has to be reworked.

I've opened the PR as a **draft** for exactly this reason — happy to adjust the approach (including switching to loop syntax) based on what you'd prefer here.
